internal/server/server_test.go

```diff
@@ -98,3 +98,48 @@ func TestSOMASServerFactoryInitialisesClients(t *testing.T) {
 		}
 	}
 }
+
+func TestRandomUniqueAssignIIGORoles(t *testing.T) {
+	iterations := 10
+	cases := []struct {
+		name  string
+		input []shared.ClientID
+	}{
+		{
+			name:  "RandomAssign 2 entries",
+			input: []shared.ClientID{shared.ClientID(0), shared.ClientID(1)},
+		},
+		{
+			name:  "RandomAssign 3 entries",
+			input: []shared.ClientID{shared.ClientID(0), shared.ClientID(1), shared.ClientID(2)},
+		},
+		{
+			name: "RandomAssign several entries",
+			input: []shared.ClientID{
+				shared.ClientID(0),
+				shared.ClientID(1),
+				shared.ClientID(2),
+				shared.ClientID(3),
+				shared.ClientID(4),
+				shared.ClientID(5)},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if len(tc.input) < 3 {
+				randomUniqueAssignIIGORoles(tc.input) // Only check for crash
+			} else {
+				for i := 0; i < iterations; i++ { // As its using random numbers. Run each test several times to minimise probablability
+					speaker, judge, president := randomUniqueAssignIIGORoles(tc.input)
+					if speaker == judge ||
+						speaker == president ||
+						judge == president {
+						t.Errorf("Speaker: %v, Judge: %v, President: %v, should all be unique but weren't", speaker, judge, president)
```

Comment:
I'd refactor as the golang playground link, then to test for uniqueness, write a utility function 

```
func lstHasUniqueClientIDs(lst []shared.ClientID) bool {
    // set to contain what we've seen so far
    s := map[shared.ClientID]interface{}
    for _, id := range lst {
        if _, ok := s[id]; ok { 
            return false
        }
        s[id] = nil
    }
    return true
}

Reply:
Thanks. I'll add this